### PR TITLE
more detail typescript definitions

### PR DESCRIPTION
**Pull request by eslym**
I rewrite the d.ts, I realize there is a lot of functions / features without documentation or having wrong documentation,
I really like how Metro 4 UI looks, hopefully there will be more contributors to maintain these projects

### CODE REVIEW

File: ts/m4q.d.ts [hunk 1]
```diff
@@ -1,328 +1,1140 @@
...
+
+        left(left: number): M4QObject<T>;
+
+        left(): number;
+
+        top(top: number): M4QObject<T>;
+
+        top(): number;
+
+        crood(): DOMRect | undefined;
```
Comment (olton) [inline ts/m4q.d.ts:1036]: `crood` => `coord`

File: ts/m4q.d.ts [hunk 2]
```diff
@@ -1,328 +1,1140 @@
...
+    isPreventedDefault: boolean;
+
+    stop(immediate?: boolean);
+}
+
+//endregion
+
+//region src/populate.js
+interface Window {
+    m4q: M4Q.M4Q;
```
Comment (olton) [inline ts/m4q.d.ts:1137]: `m4q: M4Q;` // ??

**Review by olton — COMMENTED** [review]
First of all - thanks for your work, I added two comments to your request. Secondly - please create pull request to 1.0.8 branch. Thank you.

File: ts/m4q.d.ts (hunk 2 above)
Comment (eslym) [reply ts/m4q.d.ts:1137]: the M4Q represent static functions, this allow typescript can recognize window.m4q(...)

**Comment by eslym** [comment]
> First of all - thanks for your work, I added two comments to your request. Secondly - please create pull request to 1.0.8 branch. Thank you.

noted, I will fix the typo and create a new pull request

**Pull request closed without merging**